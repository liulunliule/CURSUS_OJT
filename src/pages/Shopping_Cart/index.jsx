import React, { useEffect, useState } from "react";
import "./index.scss";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAnglesLeft, faXmark } from "@fortawesome/free-solid-svg-icons";
import { useDispatch, useSelector } from "react-redux";
import { line } from "../../assets";
import {
  deleteCartItem,
  fetchShoppingCart,
} from "../../redux/features/shoppingCartSlice";
import { toast } from "react-toastify";

const Shopping_Cart = () => {
  const dispatch = useDispatch();
  const { items, status, error } = useSelector((state) => state.shoppingCart);
  const [deleteSuccess, setDeleteSuccess] = useState(false);

  const handleDelete = async (itemId) => {
    try {
      await dispatch(deleteCartItem(itemId)).unwrap();
      setDeleteSuccess(true);
    } catch (err) {
      toast.error("Failed to delete item");
    }
  };

  useEffect(() => {
    if (status === "idle") {
      dispatch(fetchShoppingCart());
    }
  }, [status, dispatch]);

  useEffect(() => {
    if (deleteSuccess) {
      toast.success("Item deleted successfully");
      setDeleteSuccess(false);
    }
  }, [deleteSuccess]);
  return (
    <div className="Shopping_cart">
      <div className="toolbar_certification">
        <div className="container">
          <div className="row">
            <div className="col-lg-12">
              <div className="toolbar_title">
                <div className="titleleft">
                  <div className="toolbar_title_item">
                    <nav aria-label="breadcrumb">
                      <ol className="breadcrumb">
                        <li className="breadcrumb-item">
                          <Link to="/">Home</Link>
                        </li>
                        <li className="breadcrumb-item">
                          <Link to="/secondLayout/certification_center">
                            Certification Center
                          </Link>
                        </li>
                        <li
                          className="breadcrumb-item active"
                          aria-current="page"
                        >
                          Shopping Cart
                        </li>
                      </ol>
                    </nav>
                  </div>
                </div>
                <div className="titleright">
                  <Link
                    to="/secondLayout/certification_center"
                    className="blog_link"
                  >
                    <FontAwesomeIcon icon={faAnglesLeft} /> Back to
                    Certification Center
                  </Link>
                </div>
              </div>
              <div className="main_title">
                <h2>Shopping Cart</h2>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="container">
        <div className="row">
          <div className="col-lg-8">
            {status === "loading" && <p>Loading...</p>}
            {status === "failed" && <p>{error}</p>}
            {status === "succeeded" &&
              items.map((item, index) => (
                <div className="CourseInCard">
                  <Link
                    to="/course_detail/course_detail_about"
                    className="CourseInCard_img"
                  >
                    <img className="cart_img" src={item.video} alt="" />
                  </Link>
                  <div className="CourseInCard_content">
                    <div
                      className="CourseInCard_content_X"
                      onClick={() => handleDelete(item.id)}
                    >
                      <FontAwesomeIcon
                        icon={faXmark}
                        style={{
                          color: "#afafaf !important",
                          width: "15px",
                          height: "100%",
                        }}
                      />
                    </div>
                    <Link
                      to="/course_detail/course_detail_about"
                      className="CourseInCard_title"
                    >
                      {item.titilecourse}
                    </Link>
                    <Link
                      to="/detail_category_page"
                      className="CourseInCard_cate"
                    >
                      {item.typecourse}
                    </Link>
                    <div className="auth">
                      <p>
                        By <Link to="/instructor_profile">{item.author}</Link>
                      </p>
                      <div className="price">${item.price}</div>
                    </div>
                  </div>
                </div>
              ))}
          </div>
          <div className="col-lg-4">
            <div class="checkout_chk_bg stickytime">
              <div class="checkout_title">
                <h4>Total</h4>
                <img src={line} alt="" />
              </div>
              <div class="order_dt_section">
                <div class="order_title">
                  <h4>Orignal Price</h4>
                  <div class="order_price">$15</div>
                </div>
                <div class="order_title">
                  <h6>Discount Price</h6>
                  <div class="order_price">$5</div>
                </div>
                <div class="order_title">
                  <h2>Total</h2>
                  <div class="order_price5">$10</div>
                </div>
                <div class="coupon_code">
                  <p>Learn now is applied.</p>
                  <div class="coupon_input">
                    <div class="UI search coupon_input_content">
                      <div class="UI input panel-body_text_item">
                        <input
                          class="prompt srch_explore"
                          type="text"
                          name="couponcode"
                          value=""
                          id="id_coupon_code"
                          required=""
                          maxlength="6"
                          placeholder="Enter Coupon Code"
                        />
                      </div>
                      <button class="code-apply-btn" type="submit">
                        Apply
                      </button>
                    </div>
                  </div>
                </div>
                <Link to="/secondLayout/checkout_page" class="Checkout">
                  Checkout Now
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Shopping_Cart;
