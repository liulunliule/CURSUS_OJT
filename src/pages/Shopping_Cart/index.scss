.Shopping_cart{
    font-family: 'Roboto', sans-serif;
    background-color: #f8f9fa;
    padding: 0;
    margin-bottom: 5px;
    .container{
        max-width: 1200px;
    }
}

.CourseInCard {
    background: #fff;
    width: 100%;
    float: left;
    padding: 10px;
    border-radius: 3px;
    border: 1px solid #efefef;
    transition: all .2s ease-in-out;
    margin-bottom: 60px;
    margin-top: 47px;
    &_img{
        width: 40% !important;
        float: left;
        position: relative;

        img {
            width: 100%;
            border-radius: 3px;
        }
    }
    &_content{
        width: 60% !important;
        padding: 0px 15px 0px 20px;
        float: left;
        position: relative;
        &_X{
            top: 5px;
            display: inline-block;
            position: absolute;
            right: 5px;
        }
    }
    &_title{
        display: block;
        font-size: 16px;
        font-weight: 600;
        font-family: 'Roboto', sans-serif;
        color: #333;
        text-align: left;
        margin-bottom: 7px;
        line-height: 24px;
        padding-top: .5rem !important;
    }
    &_cate{
        display: block;
        font-size: 13px;
        font-weight: 400;
        font-family: 'Roboto', sans-serif;
        color: #686f7a;
        text-align: left;
        padding-top: 2px;
    }
    .auth {
        float: left;
        width: 100%;
        margin-top: 18px;
        p{    
            margin-bottom: 0;
            margin-top: 18px;
            font-size: 14px;
            font-family: 'Roboto', sans-serif;
            float: left;
            color: #686f7a;
            text-align: left;
            line-height: 24px;
        }
        a {
            color: #333;
            font-weight: 500;
        }
        .price{
            float: right;
            font-size: 18px;
            font-weight: 500;
            font-family: 'Roboto', sans-serif;
            color: #333;
            text-align: right;
        }
    }
}

.coupon{
    &_code {
        p {
            font-family: 'Roboto', sans-serif;
            padding: 20px 0 0;
            float: left;
            width: 100%;
        }
    }
    &_input{
        &_content{
            margin-top: 15px;
        }
    }
}
.UI{
    .search .prompt {
        border-radius: 3px !important;
    }
}
.code-apply-btn {
    height: 40px;
    padding: 0 20px;
    border: 0;
    /* margin-top: 45px; */
    /* margin-bottom: 20px; */
    color: #fff;
    border-radius: 3px;
    font-family: 'Roboto', sans-serif;
    font-weight: 500;
    background: #ed2a26;
    /* display: block; */
    position: absolute;
    right: 0;
}
.Checkout{
    font-size: 14px;
    margin-top: 30px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    color: #fff !important;
    background: #ed2a26;
    border-radius: 3px;
    text-align: center;
    border: 0;
    width: 100%;
    float: left;
    height: 40px;
    padding: 10px 0;
}

