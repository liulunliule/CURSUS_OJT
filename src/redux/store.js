import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
