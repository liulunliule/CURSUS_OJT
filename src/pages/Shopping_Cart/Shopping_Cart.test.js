import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Shopping_Cart from '.';

test('test Certification Test Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Shopping_Cart/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
