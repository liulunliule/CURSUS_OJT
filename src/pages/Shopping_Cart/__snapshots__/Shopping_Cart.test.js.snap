// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Certification Test Page 1`] = `
<div
  className="Shopping_cart"
>
  <div
    className="toolbar_certification"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <div
            className="toolbar_title"
          >
            <div
              className="titleleft"
            >
              <div
                className="toolbar_title_item"
              >
                <nav
                  aria-label="breadcrumb"
                >
                  <ol
                    className="breadcrumb"
                  >
                    <li
                      className="breadcrumb-item"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </li>
                    <li
                      className="breadcrumb-item"
                    >
                      <a
                        href="/secondLayout/certification_center"
                        onClick={[Function]}
                      >
                        Certification Center
                      </a>
                    </li>
                    <li
                      aria-current="page"
                      className="breadcrumb-item active"
                    >
                      Shopping Cart
                    </li>
                  </ol>
                </nav>
              </div>
            </div>
            <div
              className="titleright"
            >
              <a
                className="blog_link"
                href="/secondLayout/certification_center"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-angles-left "
                  data-icon="angles-left"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160zm352-160l-160 160c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L301.3 256 438.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 Back to Certification Center
              </a>
            </div>
          </div>
          <div
            className="main_title"
          >
            <h2>
              Shopping Cart
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-8"
      />
      <div
        className="col-lg-4"
      >
        <div
          class="checkout_chk_bg stickytime"
        >
          <div
            class="checkout_title"
          >
            <h4>
              Total
            </h4>
            <img
              alt=""
              src="line.svg"
            />
          </div>
          <div
            class="order_dt_section"
          >
            <div
              class="order_title"
            >
              <h4>
                Orignal Price
              </h4>
              <div
                class="order_price"
              >
                $15
              </div>
            </div>
            <div
              class="order_title"
            >
              <h6>
                Discount Price
              </h6>
              <div
                class="order_price"
              >
                $5
              </div>
            </div>
            <div
              class="order_title"
            >
              <h2>
                Total
              </h2>
              <div
                class="order_price5"
              >
                $10
              </div>
            </div>
            <div
              class="coupon_code"
            >
              <p>
                Learn now is applied.
              </p>
              <div
                class="coupon_input"
              >
                <div
                  class="UI search coupon_input_content"
                >
                  <div
                    class="UI input panel-body_text_item"
                  >
                    <input
                      class="prompt srch_explore"
                      id="id_coupon_code"
                      maxlength="6"
                      name="couponcode"
                      placeholder="Enter Coupon Code"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    class="code-apply-btn"
                    type="submit"
                  >
                    Apply
                  </button>
                </div>
              </div>
            </div>
            <a
              class="Checkout"
              href="/secondLayout/checkout_page"
              onClick={[Function]}
            >
              Checkout Now
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
